datafusion/substrait/src/logical_plan/consumer.rs

```diff
@@ -470,6 +471,27 @@ pub async fn from_substrait_rel(
                 "Only NamedTable reads are supported".to_string(),
             )),
         },
+        Some(RelType::Set(set)) => match set_rel::SetOp::from_i32(set.op) {
+            Some(set_op) => match set_op {
+                set_rel::SetOp::UnionAll => {
+                    assert!(!set.inputs.is_empty());
```

Comment:
It might make sense to return an error here rather than assert / panic if the input is unexpected

Reply:
Thanks @alamb! Addressed in [f30c82b](https://github.com/apache/arrow-datafusion/pull/7117/commits/f30c82bf654b297fb71819e195e967ccd99adb66).